test/compilable/dip1029.d

```diff
@@ -0,0 +1,70 @@
+
+void foo() {}
+
+void foobar() throw {}
+void bar() nothrow {}
+
+nothrow {
+    void n_t() throw;
+    void n();
+}
+
+void _t();
+void t() throw;
+
+
+struct S {
+    void foo() {
+        _t();
+        t();
+        n_t();
+    }
+}
+
+static assert(is(typeof(&S.foo) == void function()));
+
+// class inheritance
+class CA {
+    void foo() throw;
+}
+
+class CB : CA {
+    override void foo() @nogc;
+}
+
+static assert(is(typeof(&CA.foo) == void function() throw));
+static assert(is(typeof(&CB.foo) == void function() throw @nogc));
+
+
+// inferred types and attributes for lambdas
+static assert(is(typeof(() {}) == void function() pure nothrow @nogc @safe));
+static assert(is(typeof(() { throw new Exception(""); }) == void function() pure @safe));
```

Comment:
why is this not inferred as `throw` wheres the subsequent one is?

Reply:
This is inferred as `throw`, `void function() pure throw @safe` is the same as `void function() pure @safe`.

Reply:
I added a check on line 45 to prove it.